full-node/sov-ethereum/src/lib.rs

```diff
@@ -58,7 +55,7 @@ pub mod experimental {
     }
 
     pub struct Ethereum<C: sov_modules_api::Context, Da: DaService> {
-        nonces: Mutex<HashMap<RethAddress, u64>>,
+        nonce: Mutex<u64>,
```

Comment:
Let's use https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicU64.html. It is more light wait than mutex